File: core_api/internal/project/repository/dao/project.go
package dao

import "context"

type Project struct {
	ID             uint `gorm:"primaryKey"`
	UserID         uint `gorm:"index"`
	Name           string
	GameType       string
	ViewType       string
	TargetPlatform string
	Description    string
	Reference      string
	Style          string
}

type ProjectDao interface {
	CreateProject(ctx context.Context, project *Project) (uint, error)
	FindByID(ctx context.Context, id uint) (*Project, error)
	FindByUserID(ctx context.Context, userID uint) ([]*Project, error)
	Update(ctx context.Context, project *Project) error
	Delete(ctx context.Context, id uint) error
}

File: frontend/src/types/project.ts
export type Project = {
  id: string;
  name: string;
  style: string;
  gameType: string;
  platform: string;
  description: string;
};

export type ProjectSummary = Project & {
  visualStyle: string;
  visualDirection: string;
  assetCount: number;
};
